packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -3713,6 +3703,20 @@ export class SnapController extends BaseController<
     }
   }
 
+  /**
+   * Set the active state of the client. This will trigger the `onActive` or
+   * `onInactive` lifecycle hooks for all Snaps.
+   *
+   * @param active - A boolean indicating whether the client is active or not.
+   */
+  setActive(active: boolean) {
```

Comment:
This seems to be the easiest solution given that at least the extension doesn't seem to emit any events when the active status changes.

https://github.com/MetaMask/metamask-extension/blob/6e6be317110039c50f85d2768d24e9c81f3f395b/app/scripts/metamask-controller.js#L8585-L8592